### DISCUSSION TITLE
How do you properly extend a Table object?

### DISCUSSION THREAD

**Discussion by kesmit13** [ibis-project/ibis#4632] [Q&A] [answered]
I would like to add some behaviors to the Table expressions that are specific to our database, but I'm clear on what the best way is to do that. I know that I can simply plug methods on to `ir.Table` in my modules, but that would have the affect of adding them to that class for the whole Python process. I have also tried to create my own subclass of `ir.Table` which worked until I called methods like `mutate` that construct a new Table expression, but not using my subclass. I see that the Impala backend has a custom table object so it must be possible, but I'm wondering if it would require me to write my own versions of all of the supplied methods so that I would always return my custom type.

**Comment by kesmit13** [comment]
BTW, the backend we are creating is based on the SQLAlchemy backend.

**Comment by cpcloud** [comment, marked as answer]
There currently isn't a way to do this. Can you open a feature request issue if you're still interested in this feature?
